src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -486,6 +494,242 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     report_io();
 }
 
+void ov::npuw::CompiledModel::CompiledModelDesc::serialize(std::ostream& stream,
+                                                           const std::size_t& idx,
+                                                           const std::string& device) const {
+    using namespace ov::npuw::s11n;
+
+    LOG_DEBUG("Serializing CompiledModelDesc...");
+    LOG_BLOCK();
+
+    write(stream, device);
+
+    write(stream, replaced_by);
+    // FIXME: is this check enough?
+    if (replaced_by == idx) {
+        NPUW_ASSERT(compiled_model);
+        // FIXME: workaround for import/export model since import model seem to reset the file pointer
+        std::stringstream ss;
+        compiled_model->export_model(ss);
+        write(stream, ss.str());
+    }
```

Comment:
If it is the only place where you export a compiled subgraph to a blob, it is not.
In general case, non-repeating blocks _not_ forced to functions won't have `.replaced_by` at all.

Reply:
Generally, you don't need `idx` here at all.

If your CompiledModelDesc has a `compiled_model`, serialize it. That's it. No need to distinguish cases here.

Reply:
Deserialization may be interesting though

Reply:
Done